File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/adapters/recycleradapters/DevicesAdapter.java
package org.fossasia.susi.ai.device.deviceconnect.adapters.recycleradapters;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import org.fossasia.susi.ai.R;
import org.fossasia.susi.ai.device.deviceconnect.DeviceConnectPresenter;
import org.fossasia.susi.ai.device.deviceconnect.adapters.viewholders.DeviceViewHolder;
import org.fossasia.susi.ai.device.deviceconnect.adapters.viewholders.WifiViewHolder;
import org.fossasia.susi.ai.device.deviceconnect.contract.IDeviceConnectPresenter;

import java.util.List;

public class DevicesAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private List<String> itemList;
    private IDeviceConnectPresenter devicePresenter;
    private int viewCode;

    public DevicesAdapter(List<String> itemList, IDeviceConnectPresenter devicePresenter, int viewCode) {
        this.itemList = itemList;
        this.devicePresenter = devicePresenter;
        this.viewCode = viewCode;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        if (viewCode == 1) {
            View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.device_layout, parent, false);
            return new DeviceViewHolder(v, (DeviceConnectPresenter) devicePresenter);
        } else {
            View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.layout_wifi_item, parent, false);
            return new WifiViewHolder(v, (DeviceConnectPresenter) devicePresenter);
        }
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        if (holder instanceof DeviceViewHolder) {
            DeviceViewHolder viewHolder = (DeviceViewHolder) holder;
            String ssid = itemList.get(position);
            viewHolder.speakerName.setText(ssid);
        } else if (holder instanceof WifiViewHolder) {
            WifiViewHolder viewHolder = (WifiViewHolder) holder;
            viewHolder.wifiName.setText(itemList.get(position));

        }
    }

    @Override
    public int getItemCount() {
        return itemList.size();
    }
}

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/adapters/recycleradapters/DevicesAdapter.kt
package org.fossasia.susi.ai.device.deviceconnect.adapters.recycleradapters

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import org.fossasia.susi.ai.R
import org.fossasia.susi.ai.device.deviceconnect.DeviceConnectPresenter
import org.fossasia.susi.ai.device.deviceconnect.adapters.viewholders.DeviceViewHolder
import org.fossasia.susi.ai.device.deviceconnect.adapters.viewholders.WifiViewHolder
import org.fossasia.susi.ai.device.deviceconnect.contract.IDeviceConnectPresenter

class DevicesAdapter(private val itemList: List<String>, private val devicePresenter: IDeviceConnectPresenter, private val viewCode: Int) : androidx.recyclerview.widget.RecyclerView.Adapter<androidx.recyclerview.widget.RecyclerView.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): androidx.recyclerview.widget.RecyclerView.ViewHolder {
        return if (viewCode == 1) {
            val deviceLayout = LayoutInflater.from(parent.context).inflate(R.layout.device_layout, parent, false)
            DeviceViewHolder(deviceLayout, devicePresenter as DeviceConnectPresenter)
        } else {
            val layoutWifiItem = LayoutInflater.from(parent.context).inflate(R.layout.layout_wifi_item, parent, false)
            WifiViewHolder(layoutWifiItem, devicePresenter as DeviceConnectPresenter)
        }
    }

    override fun onBindViewHolder(holder: androidx.recyclerview.widget.RecyclerView.ViewHolder, position: Int) {
        if (holder is DeviceViewHolder) {
            val ssid = itemList[position]
            holder.speakerName.text = ssid
        } else if (holder is WifiViewHolder) {
            holder.wifiName.text = itemList[position]
        }
    }

    override fun getItemCount(): Int {
        return itemList.size
    }
}
